server_test.go

```diff
@@ -441,32 +441,17 @@ func checkBufferSplit(t *testing.T, buf []byte) {
 }
 
 func readTestKeysCerts() (map[string]string, error) {
-	// reads the insecure test keys and certificates in fixtures generated with:
-	// # Generate the authority key and certificate (1024-bit RSA signed using SHA-256)
-	// openssl genrsa -out cakey.pem 1024
-	// openssl req -new -x509 -key cakey.pem -out cacert.pem -days 1095 -subj "/O=Example Inc/CN=Example Certificate Authority"
-
-	// # Generate the server key and certificate, signed by the authority
-	// openssl genrsa -out serverkey.pem 1024
-	// openssl req -new -key serverkey.pem -out serverkey.csr -days 1095 -subj "/O=Example Inc/CN=localhost"
-	// openssl x509 -req -in serverkey.csr -CA cacert.pem -CAkey cakey.pem -CAcreateserial -out servercert.pem -days 1095
-
-	// # Generate a client key and certificate, signed by the authority
-	// openssl genrsa -out clientkey.pem 1024
-	// openssl req -new -key clientkey.pem -out clientkey.csr -days 1095 -subj "/O=Example Inc/CN=Veneur client key"
-	// openssl x509 -req -in clientkey.csr -CA cacert.pem -CAkey cakey.pem -CAcreateserial -out clientcert_correct.pem -days 1095
-
-	// # Generate another ca and sign the client key
-	// openssl genrsa -out wrongcakey.pem 1024
-	// openssl req -new -x509 -key wrongcakey.pem -out wrongcacert.pem -days 1095 -subj "/O=Wrong Inc/CN=Wrong Certificate Authority"
-	// openssl x509 -req -in clientkey.csr -CA wrongcacert.pem -CAkey wrongcakey.pem -CAcreateserial -out clientcert_wrong.pem -days 1095
-
+	// reads the insecure test keys and certificates in fixtures
+	// generated with: Run the testdata/_bin/generate_certs.sh
+	// script (tested on macOS Catalina) to re-generate the CA and
```

Comment:
Since not everyone has access to OS X, could we test this on Linux just to make sure there aren't any obvious errors? (Modifying the Docker container to run this and ensuring that it doesn't fail should be sufficient).

We're the ones likely to be running this script, but if we can make sure it's at least somewhat usable for others, I'd feel a lot better about this. :)

Reply:
That's fair - I mentioned the OS & version because macOS pins to a super-old version of openssl that doesn't have SAN-capable options... I've run this on the `golang:2b88645b406c` (current latest) image, and the generation script correctly generated the certs, passing tests.

I guess we could move that into the `generate` step, but I feel that it's not necessary unless go adjusts their certificate-validation again (:

Reply:
Nah, I'm fine leaving this now - just didn't want to put code out into the wild that we knew *couldn't* run on a non-proprietary OS.